idioms/str-vs-string.md

```diff
@@ -0,0 +1,89 @@
+# Using &str instead of &String
+
+## Description
+
+It is usually recommended to prefer a `&str` type over a `&String` type for arguments.  In most cases `&str` has more flexibility.  A `&String` will coerce to a `&str` type when required -- however, the opposite is not true.  
```

Comment:
Thanks for the feedback.  I wasn't comfortable to be more assertive there (though I did think about it), and I mentioned about about the size differences towards the end but not the additional layer of indirection, I'll add that as well.

In fact, @lfairy suggested that I generalize the entire article to include things like `&Vec<T>` vs `&[T]` and I wasn't sure what the proper generalization would be.  But now I think the proper generalization is: for slice-able fat pointer types.